### PR TITLE
rework and unify callbacks

### PR DESCRIPTION
**Pull request by olegrok**
Remove copy-paste code

### CODE REVIEW

File: kafka/consumer.c [hunk 1]
```diff
@@ -460,7 +332,9 @@ lua_consumer_poll_rebalances(struct lua_State *L) {
         luaL_error(L, "Usage: count, err = consumer:poll_rebalances(limit)");
 
     consumer_t *consumer = lua_check_consumer(L, 1);
-    if (consumer->event_queues == NULL || consumer->event_queues->rebalance_queue == NULL || consumer->event_queues->rebalance_cb_ref == LUA_REFNIL) {
+    if (consumer->event_queues == NULL ||
```
Comment (hackallcode) [inline kafka/consumer.c:335]: Is it possible to use LUA_RDKAFKA_POLL_FUNC here?

File: kafka/producer.c [hunk 2]
```diff
@@ -211,137 +211,9 @@ lua_producer_msg_delivery_poll(struct lua_State *L) {
...
-    }
-    lua_pushnumber(L, (double)count);
-    if (err_str != NULL) {
-        lua_pushstring(L, err_str);
-    } else {
-        lua_pushnil(L);
-    }
-    return 2;
-}
+LUA_RDKAFKA_POLL_FUNC(producer, poll_logs, LOG_QUEUE, destroy_log_msg, push_log_cb_args)
```
Comment (hackallcode) [inline kafka/producer.c:214]: Is it possible to use `LUA_RDKAFKA_POLL_FUNC ` for `lua_producer_msg_delivery_poll` function generation?

File: kafka/callbacks.c [hunk 3]
```diff
@@ -38,41 +38,37 @@ new_log_msg(int level, const char *fac, const char *buf) {
 
 void
 destroy_log_msg(log_msg_t *msg) {
-    if (msg->fac != NULL) {
```
[the remark this answers is not in the document]
Comment (olegrok) [reply kafka/callbacks.c:41]: Please see Tarantool code style https://www.tarantool.io/en/doc/latest/dev_guide/c_style_guide/#chapter-3-placing-braces-and-spaces

> Do not unnecessarily use braces where a single statement will do.

File: kafka/callbacks.c [hunk 4]
```diff
@@ -320,72 +332,55 @@ void destroy_event_queues(struct lua_State *L, event_queues_t *event_queues) {
...
+                case LOG_QUEUE:
+                    destroy_log_msg(msg);
+                    break;
+                case STATS_QUEUE:
+                    break;
+                case ERROR_QUEUE:
+                    destroy_error_msg(msg);
+                    break;
+                case REBALANCE_QUEUE: {
+                    rebalance_msg_t *rebalance_msg = msg;
```
[the remark this answers is not in the document]
Comment (olegrok) [reply kafka/callbacks.c:370]: C doesn't require explicit casts from void

[the remark this answers is not in the document]
Comment (olegrok) [reply kafka/callbacks.c:370]: Also it conforms Tarantool code-style AFAIK.

File: kafka/callbacks.h [hunk 5]
```diff
@@ -27,18 +27,27 @@ typedef struct {
     char *buf;
 } log_msg_t;
 
-log_msg_t *new_log_msg(int level, const char *fac, const char *buf);
+log_msg_t *
+new_log_msg(int level, const char *fac, const char *buf);
```
[the remark this answers is not in the document]
Comment (olegrok) [reply kafka/callbacks.h:31]: This style is used in Tarantool

File: kafka/consumer.c (hunk 1 above)
Comment (olegrok) [reply kafka/consumer.c:335]: This callback is too custom. No it's impossible.

File: kafka/producer.c (hunk 2 above)
Comment (olegrok) [reply kafka/producer.c:214]: delivery and consume callbacks have its own different logic. So, I didn't touch them.

File: kafka/callbacks.c [hunk 6]
```diff
@@ -320,72 +332,55 @@ void destroy_event_queues(struct lua_State *L, event_queues_t *event_queues) {
...
-        while (true) {
-            msg = queue_pop(event_queues->error_queue);
-            if (msg == NULL) {
-                break;
-            }
-            destroy_error_msg(msg);
-        }
-        destroy_queue(event_queues->error_queue);
-    }
     if (event_queues->delivery_queue != NULL) {
```
[the remark this answers is not in the document]
Comment (olegrok) [reply kafka/callbacks.c:341]: I think an order doesn't matter

**Review by hackallcode — APPROVED** [review]

**Pull request merged**